ocf_data_sampler/sample/base.py

```diff
@@ -5,22 +5,33 @@
 
 import logging
 import numpy as np
+import torch
+import xarray as xr
 
 from pathlib import Path
-from typing import Any, Dict, Optional, Union
+from typing import Any, Dict, Optional, Union, TypeAlias
 from abc import ABC, abstractmethod
 
+
 logger = logging.getLogger(__name__)
 
+NumpyBatch: TypeAlias = Dict[str, Any]
+TensorBatch: TypeAlias = Dict[str, Any]
+
+
 class SampleBase(ABC):
     """ 
     Abstract base class for all sample types 
     Provides core data storage functionality
     """
 
-    def __init__(self):
+    def __init__(self, data: Optional[Dict[str, Any] | xr.Dataset] = None):
         """ Initialise data container """
         logger.debug("Initialising SampleBase instance")
+        self._data = data
+    
+    def __setitem__(self, key: str, value: Any) -> None:
+        self._data[key] = value
 
     @abstractmethod
     def to_numpy(self) -> Dict[str, Any]:
```

Comment:
If we have NumpySample this could be used here 
```suggestion
    def to_numpy(self) -> NumpySample:
```